src/Scanner.js

```diff
@@ -76,6 +77,8 @@ class DefaultScanner {
     stopped = false
     /** @type {import("./Form/matching").Matching} matching */
     matching
+    /** @type Map<HTMLInputElement, HTMLFormElement> */
+    shadowInputForm = new Map()
```

Comment:
I'm not entirely clear why we need to keep the shadow forms separate from the others. Can you please explain?

Reply:
I was going to ask the same thing, can your implementation be adapted to use the existing structures?

Reply:
This is because later in `addInput` we try to get the parent form again https://github.com/duckduckgo/duckduckgo-autofill/blob/main/src/Scanner.js#L249. 

I added this to keep track if for an input, a shadow form was found. I might be able to refactor it to not add any new structures to the scanner.

Reply:
Had done a super quick one at the beginning and then I moved on and forgot about changing it :D Fixed here: https://github.com/duckduckgo/duckduckgo-autofill/pull/592/commits/5ec6e2a4503675df86d0ad0db9fd5d783f36ff4a